File: backend/model/entity/city.go
package entity

type City struct {
	Name		string	`db:"name"`
	Province	string	`db:"province"`
	Island		string	`db:"island"`
	IsAbroad	bool	`db:"is_abroad"`
}

type CreateCity struct {
	Name		string	`db:"name"`
	Province	string	`db:"province"`
	Island		string	`db:"island"`
	IsAbroad	bool	`db:"is_abroad"`
	Longitude	float64	`db:"longitude"`
	Latitude	float64	`db:"latitude"`
}

type CityList struct {
	ID		int		`db:"id"`
	Name	string	`db:"name"`
}

File: frontend/src/api/city.ts
import type { Meta } from "./user";

export interface CityData {
  cities: City[];
}

export interface City {
  id: number;
  name: string;
}

export interface SdmCity {
  data: CityListData;
}

export interface CityListData {
  cities: CityList[];
  meta: Meta;
}

export interface CityList {
  id: number;
  name: string;
  province: string;
  island: string;
  is_abroad: boolean;
  latitude: number;
  longitude: number;
}
